File: question_14.js
var strings = ["apple","banana","cherry"]
var prefix= []

function checkPrefix(prefix_array){
    let isSame = true
    console.log(prefix_array)
    for(let k=0; k<prefix_array.length; k++){
        if(k +1<prefix_array.length -1){
            if(prefix_array[k] !== prefix_array[k+1]){
                return false
            }
        } else {
            return isSame
        }
    }
}

var final_prefix = ""
for(let i =0;i <strings.length; i++){
    let prefixes = []
    for(let j=0; j<strings.length; j++){
        prefixes.push(strings[j][i])
    }
    if(!checkPrefix(prefixes)){
        console.log("loop is broken")
        break
    } else {
        console.log("prefixes: "+prefixes)
        final_prefix +=prefixes[0]
        prefix.push(prefixes)
        console.log("prefix: "+ prefix)
    }
}

console.log(prefix)

console.log("Final Prefix: "+final_prefix)

File: question_14.py
# question - 14 (Longest common prefix) [EASY]

words = input("Enter the words: ").split(" ")
print(words)


def check_prefix(prefix_array):
    is_same = True
    for k in range(len(prefix_array) -1):
        if k + 1 < len(prefix_array):
            if prefix_array[k] != prefix_array[k + 1]:
                return False

    return is_same


def common_prefix(strings):
    final_prefix = ""
    smallest_string = min(len(word) for word in strings)
    print(smallest_string)

    for j in range(smallest_string):
        prefix_chars = []
        for i in range(len(strings)):
            prefix_chars.append(strings[i][j])
        if not check_prefix(prefix_chars):
            break
        else:
            final_prefix += prefix_chars[0]

    return final_prefix

print(common_prefix(words))